homeassistant/components/nextdns/__init__.py

```diff
@@ -175,14 +175,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     # Independent DataUpdateCoordinator is used for each API endpoint to avoid
     # unnecessary requests when entities using this endpoint are disabled.
     for coordinator_name, coordinator_class, update_interval in COORDINATORS:
-        hass.data[DOMAIN][entry.entry_id][coordinator_name] = coordinator_class(
-            hass, nextdns, profile_id, update_interval
-        )
-        tasks.append(
-            hass.data[DOMAIN][entry.entry_id][
-                coordinator_name
-            ].async_config_entry_first_refresh()
-        )
+        coordinator = coordinator_class(hass, nextdns, profile_id, update_interval)
+        tasks.append(coordinator.async_config_entry_first_refresh())
+        hass.data[DOMAIN][entry.entry_id][coordinator_name] = coordinator
```

Comment:
This is actually not doing what you think it does...

Reply:
🧐 you are right ... d201e837201b83b16a77114eb4a9f45a875d9637 should correct it 🙂